Add `pathdata` type and `@pathdata` macro (#12067)
slint-ui/slint: docs/astro/src/content/docs/reference/primitive-types.mdx

</SlintProperty>

## Paths
### pathdata

> <SlintProperty propName="pathdata" typeName="pathdata" defaultValue='empty pathdata'>

Reviewer: ```suggestion
<SlintProperty propName="pathdata" typeName="pathdata" defaultValue='an empty pathdata'>
```

> The `pathdata` type is a reference to a SVG path commands.

Reviewer: Minor change, but:

```suggestion
A `pathdata` value is a compiled series of SVG path commands.
```

> A `pathdata` value can be constructed from a string literal with the `@pathdata("...")` construct.

Reviewer: ```suggestion
A `pathdata` value can be constructed at compile-time with the `@pathdata("...")` macro, or you can implicitly cast from a string to construct it at runtime.
```
